### PR TITLE
feat(forge): broadcast(uint256) and startBroadcast(uint256) overloads for private key

### LINKED ISSUE #3099: Add broadcast(uint256) and startBroadcast(uint256) overloads for private key
### Component

Forge

### Describe the feature you would like

As per discussion in #3047, we should also allow the broadcast cheatcodes to directly take in a private key to use for signing subsequent transactions in scripts.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by devanonon**
## Motivation

Closes #3099.

Discussion @ #3047.

Allows one to specify the private key to use for broadcasting transactions in a script.

## Solution

Example usage:
```solidity
vm.startBroadcast(vm.envUint("PRIVATE_KEY"));
// broadcast all the things
vm.stopBroadcast();
```
or:
```solidity
string memory mnemonic = vm.envString("MNEMONIC");
uint256 deployerPrivateKey = vm.deriveKey(mnemonic, 0);

vm.broadcast(deployerPrivateKey);
// broadcast one thing
```

### CODE REVIEW

File: evm/src/executor/inspector/cheatcodes/env.rs [hunk 1]
```diff
@@ -62,6 +68,35 @@ fn broadcast(
...
+
+    let mut bytes: [u8; 32] = [0; 32];
+    private_key.to_big_endian(&mut bytes);
+
+    let key = SigningKey::from_bytes(&bytes).map_err(|err| err.to_string().encode())?;
+    let wallet = LocalWallet::from(key).with_chain_id(chain_id.as_u64());
+
+    state.script_wallets.push(wallet.clone());
+
+    let origin = wallet.address();
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/env.rs:95]: can swap these lines to get rid of the clone?

File: evm/src/executor/inspector/cheatcodes/env.rs [hunk 2]
```diff
@@ -62,6 +68,35 @@ fn broadcast(
     Ok(Bytes::new())
 }
 
+fn broadcast_key(
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/env.rs:71]: please add brief docs

**Review by mattsse — CHANGES_REQUESTED** [review]
ty,

just a smol nit

**Comment by devanonon** [comment]
@mattsse thanks, both comments addressed now

**Review by onbjerg — APPROVED** [review]

**Pull request merged**